api4/post.go

```diff
@@ -185,6 +185,29 @@ func getPostsForChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 		mlog.Error("Failed to prepare posts for getPostsForChannel response", mlog.Any("err", err))
 	}
 
+	postListLength := len(clientPostList.Order)
+
+	var nextPostId string
+	if c.Params.Page == 0 {
+		nextPostId = beforePost
+	} else if postListLength < c.Params.PerPage {
+		nextPostId = ""
```

Comment:
2. If you're not getting the first page and less than a page of posts is returned, both the previous and next post IDs will be empty. For example, if you request `?before=post5&page=0&per_page=3` (and fix the first case), you'd get
    ```
    {
        next_post_id: "post5",
        order: ["post4", "post3", "post2"],
        prev_post_id: "post1"
    }
    ```
    but then when you get the next page with `?before=post5&page=1&per_page=3`, you'll get
    ```
    {
        next_post_id: "", // This should be "post2", but it's empty
        order: ["post1"],
        prev_post_id: ""
    }
    ```